src/concolic/concolic_wasm_ffi.ml

```diff
@@ -69,6 +69,19 @@ module M :
         let n = Float64.of_bits n in
         (F64 n, Value.pair n (Expr.mk_symbol sym)) )
 
+  let symbol_bool () : Value.int32 Choice.t =
+    Choice.with_new_symbol (Ty_bitv 32) (fun sym forced_value ->
+        let n =
+          match forced_value with
+          | None -> Int32.logand 1l (Random.bits32 ())
+          | Some (Num (I32 n)) -> n
+          | _ -> assert false
+        in
+        let sym_expr =
+          Expr.make (Cvtop (Ty_bitv 32, Zero_extend 31, Expr.mk_symbol sym))
```

Comment:
```suggestion
    Choice.with_new_symbol (Ty_bool) (fun sym forced_value ->
        let b =
          match forced_value with
          | None -> Random.bool ()
          | Some True -> true
          | Some False -> false
          | _ -> assert false
        in
        let n = V.Bool.int32 b in
        (I32 n, Value.(Bool.int32 (pair b (Expr.mk_symbol sym)))))
```
You could probably try to do something similar to what you did on the symbolic version here. (I don't know if this code compiles)

Reply:
Thanks for the proposition, it compiles and works :)